### PR TITLE
Ciam 5756 seat admin meta auth z

### PR DESCRIPTION
**Pull request by merlante**
## Describe your changes

- Extracting orgID and isOrgAdmin from auth token in addition to subjectID and setting in context.
- Add logic to seat admin grpc Server methods to return unauthorized when requestor is not an org admin for the org they are intending to assign seats for. 

### Note

- As is, this change will break our smoke test, since the authenticated user running the smoke test must be an org admin for the "o1" org in the environment (i.e. stage/prod) we're running the smoke test against. This is because the smoke test calls seat admin endpoints.

## Ticket reference (if applicable)
Fixes #CIAM-5756.

## Checklist

* [x] Are the agreed upon acceptance criteria fulfilled?

* [x] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?

* [x] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [x] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [x] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: api/grpc/interceptor/AuthnInterceptor.go [hunk 1]
```diff
@@ -241,3 +255,15 @@ func getBearerTokenFromContext(ctx context.Context) string {
 	}
 	return ""
 }
+
+func requestorOrg(token jwt.Token) (orgID string, isOrgAdmin bool) {
```
Comment (DGuhr) [inline api/grpc/interceptor/AuthnInterceptor.go:259]: Just a nitpick: Maybe rename `getOrgNameAndAdminStatusForRequestor`?

File: bootstrap/app_test.go [hunk 2]
```diff
@@ -75,19 +71,17 @@ func TestCheckReturnsTrueWhenUserAuthorized(t *testing.T) {
...
-		`{"subject": "not_authorized", "operation": "access", "resourcetype": "license", "resourceid": "o1/smarts"}`))
+	resp, err := http.DefaultClient.Do(post("/v1alpha/check", "system", "no_particular_org", false, `{"subject": "not_authorized", "operation": "access", "resourcetype": "license", "resourceid": "o1/smarts"}`))
 
 	assert.NoError(t, err)
 
 	assertJSONResponse(t, resp, 200, `{"result": %t, "description": ""}`, false)
 }
 
-func TestAssignLicenseReturnsSuccess(t *testing.T) {
+func TestAssignSeatReturnsSuccess(t *testing.T) {
```
Comment (DGuhr) [inline bootstrap/app_test.go:81]: good catch!

**Review by DGuhr — APPROVED** [review]
Just a few minor comments, feel free to change or not. All in all LGTM 👍

**Review by DGuhr — APPROVED** [review]

File: testenv/FakeIdp.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func HostFakeIdp() {
 }
 
 // SubjectIDToToken creates a new jwt token for a given SubjectID
-func SubjectIDToToken(subject string) string {
+func SubjectIDToToken(subject string, orgID string, isOrgAdmin bool) string {
```
Comment (wscalf) [inline testenv/FakeIdp.go:92]: Maybe CreateToken or something since it's not a conversion from SubjectID anymore?

**Review by DGuhr — APPROVED** [review]

**Comment by Rajagopalan-Ranganathan** [comment]
Thanks for the quick review and feedback!

**Pull request merged**